### PR TITLE
Added the lock item

### LINKED ISSUE #42: Add lock that `/data merge` locks block entities
Similar to the "locking key" from peculiar pieces, merges a standard `Lock:"glowcase_lock"` and shows action bar messages like 

`minecraft:chest is now locked`

`minecraft:chest is now unlocked`

split from #11

### PR DESCRIPTION
**Pull request by Chailotl**
Closes #42

### CODE REVIEW

File: src/main/java/dev/hephaestus/glowcase/item/LockItem.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+public class LockItem extends Item {
+	public LockItem(Settings settings) {
+		super(settings);
+	}
+
+	@Override
+	public ActionResult useOnBlock(ItemUsageContext context) {
+		World world = context.getWorld();
+		if (world.isClient || !(world.getBlockEntity(context.getBlockPos()) instanceof LockableContainerBlockEntity be)) return ActionResult.PASS;
```
Comment (sleepingdragoninn) [inline src/main/java/dev/hephaestus/glowcase/item/LockItem.java]: can we ensure the player is in creative mode here?

File: src/main/java/dev/hephaestus/glowcase/mixin/PlayerEntityMixin.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+public abstract class PlayerEntityMixin {
+	@Shadow
+	public abstract ItemStack getEquippedStack(EquipmentSlot slot);
+
+	@ModifyReturnValue(
+		method = "shouldCancelInteraction",
+		at = @At(value = "RETURN")
+	)
+	private boolean directLockInteraction(boolean original) {
+		return getEquippedStack(EquipmentSlot.MAINHAND).getItem().equals(Glowcase.LOCK_ITEM.get()) || original;
```
Comment (sleepingdragoninn) [inline src/main/java/dev/hephaestus/glowcase/mixin/PlayerEntityMixin.java:22]: What's this mixin for?

**Review by sleepingdragoninn — APPROVED** [review]
Looking good! Just two questions

File: src/main/java/dev/hephaestus/glowcase/mixin/PlayerEntityMixin.java (hunk 2 above)
Comment (sleepingdragoninn) [reply src/main/java/dev/hephaestus/glowcase/mixin/PlayerEntityMixin.java:22]: ah i see, it lets you lock chests without needing to sneak like you would to place a sign - makes sense!

File: src/main/java/dev/hephaestus/glowcase/item/LockItem.java (hunk 1 above)
Comment (Chailotl) [reply src/main/java/dev/hephaestus/glowcase/item/LockItem.java]: That would be a good idea, wouldn't want guests running around with this item ^_^;

Comment (Chailotl) [reply src/main/java/dev/hephaestus/glowcase/item/LockItem.java]: Resolved in b3cdfa0

**Pull request merged**